learning/01_Exercises/Lab 7 to 9/domain/validator.py

```diff
@@ -0,0 +1,73 @@
+class ValidatorError(Exception):
+    pass
+
+
+class validate:
+    def checkWord(self, word):
+        # The title must contain only letters and spaces. It returns True or False.
+        allLetters = ['a', 'b', 'c', 'd', 'e', 'f', 'g', 'h', 'i', 'j', 'k', 'l', 'm', 'n', 'o', 'p', 'q', 'r', 's',
+                      't', 'u', 'v', 'w', 'x', 'y', 'z', " ", "."]
+        unwantedLetters = [letter for letter in word if letter.lower() not in allLetters]
+        trueOrFalse = False if len(unwantedLetters) > 0 else True
+        return trueOrFalse
+
+    def checkID(self, ID, idList):
+        trueOrFalse = False if ID not in idList else True
+        return trueOrFalse
+
+
+class movieValidator(validate):
+    def checkAvailability(self, availability):
+        trueOrFalse = True if availability == "Available" else False
+        return trueOrFalse
+
+    def checkGenre(self, genre):
+        # This method checks if the genre given by the user exists.
+        movieGenres = ["Action", "Crime", "Drama", "Fantasy", "Horror",
+                       "Comedy", "Romance", "Science Fiction", "Sports",
+                       "Thriller", "Mystery", "War", "Western"]
+        trueOrFalse = True if genre in movieGenres else False
```

Comment:
Name of this var is weird `trueOrFalse` doesn't say much. I would name it `genre_is_valid` or something similar. Also `True if genre in movieGenres else False` can be written as `genre in movieGenres` so `genre_is_valid = genre in movieGenres` ( you use this all over this file)

Reply:
Rezolvat.